src/App.tsx

```diff
@@ -1,203 +1,57 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useState, useEffect, useContext } from 'react';
+import { getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Main } from './components/Main';
+import { ErrorContext, TodosContext } from './components/TodoContext';
+import { Status } from './types/Status';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const { list, setList } = useContext(TodosContext);
+  const { errorMessage, setErrorMessage } = useContext(ErrorContext);
+  const [query, setQuery] = useState<Status>(Status.All);
+
+  useEffect(() => {
+    getTodos()
+      .then(setList)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        setTimeout(() => setErrorMessage(''), 3000);
```

Comment:
Just use ``setErrorMessage('')`` because your TodosContext already handle errors

Reply:
tests fail without it

Reply:
Then it's better to create an error handler with error message as parameter which can show this error for 3s and close it. In future you can reuse this handler to display different errors in different components

Reply:
> Just use `setErrorMessage('')` because your TodosContext already handle errors

Good point, create a handler that will set an error and a set timeout that will remove it in 3 seconds, because you should do the same in other cases in the future tasks

Reply:
> > Just use `setErrorMessage('')` because your TodosContext already handle errors
> 
> Good point, create a handler that will set an error and a set timeout that will remove it in 3 seconds, because you should do the same in other cases in the future tasks

Not fixed